src/QMCHamiltonians/ECPComponentBuilder.2.cpp

```diff
@@ -103,8 +103,15 @@ void ECPComponentBuilder::buildSemiLocalAndLocal(std::vector<xmlNodePtr>& semiPt
       aAttrib.add(lstr, "l");
       aAttrib.add(rc, "cutoff");
       aAttrib.put(cur_vps);
-      rmax  = std::max(rmax, rc);
-      int l = angMon[lstr];
+      rmax = std::max(rmax, rc);
+      int l;
+      if (angMon.find(lstr) == angMon.end())
+      {
+        app_error() << "requested angular momentum " << lstr << " not available.\n";
+        APP_ABORT("ECPComponentBuilder::buildSemiLocalAndLocal");
+      }
+      else
+        l = angMon[lstr];
```

Comment:
Avoid else. Put
```
int l = angMon[lstr];
```
Compiler is not smart. It may complains access to uninitialized l.

Reply:
Good point Ye.

Reply:
done